bindings/matrix-sdk-crypto-js/src/verification.rs

```diff
@@ -906,6 +906,15 @@ impl VerificationRequest {
         self.inner.timed_out()
     }
 
+    /// The number of milliseconds remaining before this verification flow times
+    /// out.
+    ///
+    /// Returns zero if the time has already passed.
+    #[wasm_bindgen(js_name = "timeRemainingMillis")]
+    pub fn time_remaining_millis(&self) -> f64 {
+        self.inner.time_remaining().as_millis() as f64
```

Comment:
This cast makes me uncomfortable. We are casting a `u128` to a `f64`. Casting to floats seems to always be a bad idea :-p.

What about returning a `u64`, or even a `u32`? There is `TryFrom<u128>` implementations for `u64` and `u32` as far as I reckon:

```rust
pub fn time_remaining_millis(&self) -> u64 {
    // SAFETY: `unwrap`ing is safe because the remaining time is always
    // lower than 2^64, making this `unwrap` unreachable.
    self.inner.time_remaining().as_millis().try_into().expect("failed to cast `time_remaining` to `u64`")
}
```

Thoughts?
It's

Reply:
Well, I felt the same initially, so I tried returning a u64, but it got turned into a BigInt, which would have meant doing annoying conversions in the application. It's possible I was doing something wrong, but fundamentally, for better or worse, it's idiomatic to represent numbers as floating point in Javascript.

Given that, I think it makes sense for the bindings to cast to float.

In this case (a) we know it's going to be in the range [0,600000] which an f64 is perfectly capable of representing without loss of precision; (b) even that *wasn't* the case, we'd end up losing precision rather than range, and we don't really care about the difference between (a long time) and (a long time minus a few milliseconds).

Reply:
And what about returning a `u32` :-p? It's not a `BigInt` in JavaScript right?

Otherwise, fine, let's go with the actual solution.

Reply:
I didn't try a u32 tbh, but one way or the other we *want* to cast to a float for Javascript; we can either go via a u32 with fiddly error handling and adds no benefit in range or precision; or go straight to f64 which just works.